common/tools/dev-tool/src/util/testUtils.ts

```diff
@@ -0,0 +1,44 @@
+import { isProxyToolActive } from "./testProxyUtils";
+import concurrently from "concurrently";
+import { createPrinter } from "./printer";
+
+const log = createPrinter("preparing-proxy-tool");
+
+async function shouldRunProxyTool(): Promise<boolean> {
+  const mode = process.env.TEST_MODE;
+  if (mode === "live") {
+    return false; // No need to start the proxy tool in the live mode
+  } else {
+    try {
+      await isProxyToolActive();
+      // No need to run a new one if it is already active
+      // Especially, CI uses this path
+      log.info(
+        `Proxy tool seems to be active, not attempting to start the test proxy at http://localhost:5000 & https://localhost:5001.\n`
```

Comment:
Same parametrized ports question.

Reply:
Logged an issue for this at #15829, will tackle in a future PR.